src/recognize.rs

```diff
@@ -0,0 +1,135 @@
+// Copyright 2017 Google Inc. All rights reserved.
```

Comment:
The intent is for it to be covered by the same licenses as the repo. I think the best way to do this (while preserving the Google copyright for the new file) is to just add boilerplate so there's no confusion.